stdlib/public/Distributed/DistributedActor.swift

```diff
@@ -100,9 +103,9 @@ extension CodingUserInfoKey {
 @available(SwiftStdlib 5.5, *)
 extension DistributedActor {
   nonisolated public init(from decoder: Decoder) throws {
-    guard let transport = decoder.userInfo[.actorTransportKey] as? ActorTransport else {
+    guard let transport = decoder.userInfo[.actorTransportKey] as? Transport else {
```

Comment:
Hmmm... I see, this is more restrictive but I think it'll be fine -- realistically people should not be mixing completely random transports and sending an actor from one transport to another transport, I suppose...

Reply:
Oh, that's a good point, this might be a bug! We probably need `AnyActorTransport` to be `Codable` over top of the underlying transport, so we can decide it.

Reply:
Nah, transports are not intended to be Codable; This change seems fine.

I was thinking about a weird use-case that to be honest we don't have to support IMHO -- "mixing transports" which is pretty complicated and I don't think we really need it in any real use-case to be honest.

Here is the imagined problem:
- you could receive a `websocket://12.3.4:####/greeter` identified actor... the `WebSocketTransport` decodes this fine...
- you are ALSO part of the cluster, and send the same ref to some cluster actor...
- encoding just works, it's Codable by the websocket address...
- the recipient system, attempts to decode it
  - puts self into `.actorTransportKey`, that's the Cluster transport
  - that as? will fail, since this is `Cluster` and not `WebSocket` transport after all

To be honest this is likely  _good_, you can't just randomly share refs like that as the connections and forwarding take a role in all that too... I think this is okey to keep as such runtime `"failed to decode websocket:/// actor using Cluster transport" it makes total sense and is easily fixed by developers with a "forwarder" actor if they need such chain of messages.


Why all of this doesn't really matter:
- the connection of course didn't "move" with the actor as we sent it over like that...
- so it does not make sense for that other system to resolve the "actual connection backed actor", since it can't, the connection is on the first node -- that accepted that websocket actor.

So yeah, such weird thing could happen, but honestly I think it's a weird edge case, and by means of our transports being explicit in the types -- we'd see that "huh, why am I sending this websocket actor to the cluster, does this make sense?" One can spin up a simple "forwarder" actor if we needed to support that.